localstack/services/events/provider.py

```diff
@@ -71,6 +73,27 @@ def on_before_stop(self):
     def get_store(context: RequestContext) -> EventsStore:
         return events_stores[context.account_id][context.region]
 
+    def test_event_pattern(
+        self, context: RequestContext, event_pattern: EventPattern, event: String
+    ) -> TestEventPatternResponse:
+
+        # https://docs.aws.amazon.com/eventbridge/latest/APIReference/API_TestEventPattern.html
+        # Test event pattern uses event pattern to match against event.
+        # So event pattern keys must be in the event keys and values must match.
+        # If event pattern has a key that event does not have, it is not a match.
+        evt_pattern = json.loads(str(event_pattern))
+        evt = json.loads(str(event))
+
+        for key in evt_pattern.keys():
+            if key not in evt.keys():
+                return TestEventPatternResponse(Result=False)
+
+        for key, values in evt_pattern.items():
+            if evt[key] not in values:
+                return TestEventPatternResponse(Result=False)
```

Comment:
nit: we could only iterate once over the dict:
```suggestion
        for key, values in evt_pattern.items():
            if key not in evt or evt[key] not in values:
                return TestEventPatternResponse(Result=False)
```

We could even make use of `any`:
```suggestion
        if any(key not in evt or evt[key] not in values for key, values in evt_pattern.items()):
            return TestEventPatternResponse(Result=False)
```

Reply:
Yep nice suggestions, I'll improve on the any suggestion since has that functional aspect I appreciate.